plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePartitionManager.java

```diff
@@ -200,8 +223,23 @@ public HiveTableHandle applyPartitionResult(HiveTableHandle handle, HivePartitio
 
     public List<HivePartition> getOrLoadPartitions(SemiTransactionalHiveMetastore metastore, HiveIdentity identity, HiveTableHandle table)
     {
+        // In case of partitions not being loaded, their permissible values are specified in `HiveTableHandle#getCompactEffectivePredicate,
+        // so we do an intersection of getCompactEffectivePredicate and HiveTable's enforced constraint
+        TupleDomain<ColumnHandle> summary = table.getEnforcedConstraint().intersect(
+                table.getCompactEffectivePredicate()
+                        .transformKeys(ColumnHandle.class::cast));
         return table.getPartitions().orElseGet(() ->
-                getPartitionsAsList(getPartitions(metastore, identity, table, new Constraint(table.getEnforcedConstraint()))));
+                getPartitionsAsList(
+                        getPartitions(
+                                metastore,
+                                identity,
+                                table,
+                                new Constraint(summary))));
+    }
+
+    public boolean hasPartitionsLessThanThreshold(HivePartitionResult partitionResult)
+    {
+        return partitionResult.getPartitionNames().orElse(ImmutableList.of()).size() <= maxPartitions;
```

Comment:
as above, the `partitionResult.getPartitionNames().orElse(ImmutableList.of()).size()` is tricky hack.
Split into separate conditions.

For the usage above, i'd inline.
I see that this partition handling leaked into `io.trino.plugin.hive.HiveMetadata#getTableProperties`
if we can't avoid it, let's find a better name for the method (currently it's "what threshold??")
Something like `canPartitionsBeLoaded`?

Reply:
i see you renamed to `canPartitionsBeLoaded`